omegaconf/base.py

```diff
@@ -429,72 +424,61 @@ def _resolve_interpolation_from_parse_tree(
     def _resolve_node_interpolation(
         self,
         inter_key: str,
-        throw_on_missing: bool,
-        throw_on_resolution_failure: bool,
-    ) -> Optional["Node"]:
+    ) -> "Node":
         """A node interpolation is of the form `${foo.bar}`"""
-        root_node, inter_key = self._resolve_key_and_root(inter_key)
-        parent, last_key, value = root_node._select_impl(
-            inter_key,
-            throw_on_missing=throw_on_missing,
-            throw_on_resolution_failure=throw_on_resolution_failure,
-        )
+        try:
+            root_node, inter_key = self._resolve_key_and_root(inter_key)
+        except ConfigKeyError as exc:
+            raise InterpolationKeyError(str(exc))
```

Comment:
When you are translating exceptions like this:
1. If the exception has a multi-line message (like normal OmegaConf exceptions) it will get up repeated multiple times.
2. You are losing the stack trace.

Something like this will retain the current stack trace and also chain the original exception (not sure both makes sense).
I think the most important thing is to retain the stack trace, otherwise it's very difficult to debug.
```python
raise InterpolationKeyError("ConfigKeyError while resolving interpolation").with_traceback(sys.exc_info()[2]) from exc
```

This happens in multiple places. I will not comment on all of them.

Reply:
Ah, thanks, I was indeed a bit annoyed to lose the stack trace, but the quick search I did only uncovered some ugly stuff -- I should have kept searching.

This should be all fixed in 5718bf0

Regarding point (1), I don't think it's an issue here because the exceptions I'm catching haven't been processed by format_and_raise() yet (this will happen after they are re-raised). At least I didn't see any issue with repetitions.

Exception chaining didn't make a difference in all places I tried it. I suspect that it's because of what's happening in format_and_raise() (I didn't look very closely) -- it must ignore it somehow. So I stuck to `with_traceback()` only.